tests/nexus_workflow_test.go

```diff
@@ -1505,6 +1506,132 @@ func (s *NexusWorkflowTestSuite) TestNexusAsyncOperationWithNilIO() {
 	}
 }
 
+func (s *NexusWorkflowTestSuite) TestNexusResetWithCallback() {
```

Comment:
I would move this test to `callbacks_test.go`, there's no need to test the caller workflow, this will give you more control over handling the callbacks.

Reply:
Ack